certora/specs/RevertsERC20.spec

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+methods {
+    function balanceOf(address) external returns uint256 envfree;
+    function allowance(address, address) external returns uint256 envfree;
+
+    // Avoids checking delegation related overflows.
+    function _._moveDelegateVotes(address, address, uint256) internal => CONSTANT;
+}
+
+// Check that transfers with zero address or too large amounts revert.
+rule transferReverts(env e, address to,uint256 amount) {
+    // Safe require as transfer is a non-payable function.
+    require e.msg.value == 0;
+
+    uint256 balanceOfSenderBefore = balanceOf(e.msg.sender);
+
+    transfer@withrevert(e, to, amount);
+    assert !lastReverted <=> e.msg.sender != 0 && to !=0 && balanceOfSenderBefore >= amount;
+}
+
+
+// Check that transfersFrom with zero address or too large allowances revert.
```

Comment:
```suggestion
// Check that transfersFrom with zero address or too large amount revert.
```

Reply:
Fixed in https://github.com/morpho-org/morpho-token-upgradeable/pull/98/commits/0530fa60e78c0e504becd3b20ac3014a65a07379.